trinity/extensibility/asyncio.py

```
@@ -49,6 +49,8 @@ def get_subprocess_kwargs(self) -> Optional[SubprocessKwargs]:
                     proc.send_signal(signal.SIGTERM)
                 finally:
                     raise err
+        if proc.error:
```

I'm not 100% sure but I think it is possible for this line to raise an `AttributeError`

https://github.com/ethereum/asyncio-run-in-process/blob/74f11319739c64c0fe8ff7c2aebee3ecf5a125e8/asyncio_run_in_process/process.py#L181-L189

It hints at a missing API which is the ability to check for an error.  Maybe something like `Process.did_error`?  or to just fix/adjust the `Process.error` property to return a `None` value...

---

ah good catch -- i was just looking at the fn signature...

i agree about being able to check for errors...

in this case, i think we can get away with just `await .proc.wait_result()`, which is not very informative but should raise the exception if there is one...